### PR TITLE
Parse email files "smtp mail" type

### PR DESCRIPTION
**Pull request by glicht**
## Status
Ready

## Related Issues
related to: https://github.com/demisto/etc/issues/15277

## Description
A fix to work with eml files which have an "smtp mail" type

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests: unit test run locally
- [x] Code Review

## Additional changes
Also added package_extractor.py script to extract python code from yml file.

### CODE REVIEW

File: package_extractor.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+import yaml
+import argparse
+from io import open
+
+INTEGRATION = 'integration'
+SCRIPT = 'script'
+
+
+def extract_code(yml_path, output_path, demisto_mock, commonserver=None, yml_type=None):
```
Comment (ronykoz) [inline package_extractor.py]: I would have splitted this into couple of functions or just added some blank lines for some visual/logical separation, you have a setup stage and then extraction, and creation stages here

File: package_extractor.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        yml_data = yaml.safe_load(yml_file)
+        script = yml_data['script']
+        if yml_type == INTEGRATION:  # in integration the script is stored at a second level
+            script = script['script']
+    with open(output_path, 'w', encoding='utf-8') as code_file:
+        if demisto_mock:
+            code_file.write(u"import demistomock as demisto\n")
+        if commonserver:
+            code_file.write(u"from CommonServerPython import *\n")
+        if type(script) == str:  # pyyaml may return either a unicode or str depending upon the content of script
```
Comment (ronykoz) [inline package_extractor.py]: lets move this to the first `with`

File: Scripts/ParseEmailFiles/parse_email_files_test.py [hunk 3]
```diff
@@ -22,3 +24,29 @@ def test_msg_utf_encoded_subject():
...
+                    {'info': 'SMTP mail, UTF-8 Unicode text, with CRLF terminators'}}]
+        else:
+            raise ValueError('Unimplemented command called: {}'.format(name))
+
+    mocker.patch.object(demisto, 'args', return_value={'entryid': 'test'})
+    mocker.patch.object(demisto, 'executeCommand', side_effect=executeCommand)
+    mocker.patch.object(demisto, 'results')
+    # validate our mocks are good
+    assert demisto.args()['entryid'] == 'test'
+    # assert demisto.executeCommand('getFilePath', {})[0]['Type'] == entryTypes['note']
```
[the remark this answers is not in the document]
Comment (glicht) [reply Scripts/ParseEmailFiles/parse_email_files_test.py]: will do this on separate PR where will be trying to run the unit tests as part of the build

**Comment by glicht** [comment]
The package_extractor.py is a small script I need for getting the CommonServerPython for moving to a mode where we run unit tests as part of the build. I was also using this locally. Let's just leave this for now until we progress a bit. I am planning to do a separate PR which includes code for running the unit tests.

**Pull request merged**